### ISSUE TITLE
FalsePositive MissUsingForeachInspection

### ISSUE THREAD

**Issue by puggan** [kalessil/phpinspectionsea#1838]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | phpInspectionsUltimate-mp (2021.5) |
| Language level | PHP 8.0 |

Current behaviour
--
MissUsingForeachInspection asks me to use array_values(),
if `$hotelCombinationRow->resid` was constant, that could have worked, but as that key is dynamic, it's not posible (as far as I know) to replace this foreach with an array_values().

![image](https://user-images.githubusercontent.com/866668/176481222-a2c28de4-37f1-4968-836d-517479d215af.png)

    $hotelCombinationRows = DB::select($query);
    foreach ($hotelCombinationRows as $hotelCombinationRow) {
        $resor[$hotelCombinationRow->resid]->combinatoins[] = $hotelCombinationRow;
    }

Expected behaviour
--
No MissUsingForeachInspection

Environment details
---

Information from Help - About dialog

PhpStorm 2022.2 EAP
Build #PS-222.3153.10, built on June 22, 2022
Licensed to PhpStorm EAP user: <me>
Expiration date: July 22, 2022
Runtime version: 17.0.3+7-b469.12 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 2010M
Cores: 8
Registry:
    run.processes.with.pty=TRUE

Non-Bundled Plugins:
    com.kalessil.phpStorm.phpInspectionsUltimate-mp (2021.5)
    de.espend.idea.php.toolbox (6.1.0)
    de.espend.idea.php.annotation (8.2.1)
    NEON support (0.5.2)
    com.jetbrains.plugins.jade (222.3153.10)
    izhangzhihao.rainbow.brackets (6.25)
    fr.adrienbrault.idea.symfony2plugin (2022.1.230)
    ru.adelf.idea.dotenv (2022.2)

**Comment by JacekAndrzejewski** [comment]
On every loop you just push one element to `$resor[$hotelCombinationRow->resid]->combinatoins` array, you can replace it with `array_values` call if that array had no elements before the loop.

This should work the same for empty combinations array (which probably is empty as you made a typo in the name)
```
$hotelCombinationRows = DB::select($query);
$resor[$hotelCombinationRow->resid]->combinatoins= array_values($hotelCombinationRows);
```

**Comment by jdreesen** [comment]
I just stumbled upon the same FP with this code:

![grafik](https://user-images.githubusercontent.com/424602/189085145-69ab883b-3bc3-47a7-af99-f0721d0aa905.png)

```php
$result = [];
foreach ($this->elements as $element) {
    $result[$element->getType()][] = $element;
}
```

I doubt that I can just use `array_values()` instead, as it's a grouping operation.
